File: test_cases/c/gpt/02.c
#include <assert.h>
#include <stdbool.h>

bool isMatch(char* s, char* p);

int main(void) {
    assert(isMatch("aa", "a") == false);
    assert(isMatch("aa", "a*") == true);
    assert(isMatch("ab", ".*") == true);
    assert(isMatch("aab", "c*a*b") == true);
    assert(isMatch("aaaaaaaaaaaaaaaaaaaa", "a*a*a*a*a*a*a*a*a*a*") == true);

    return 0;
}

File: test_cases/python/gpt/02.py
def run_tests(solution_module):
    f = solution_module.isMatch
    assert f('aa', 'a') is False
    assert f('aa', 'a*') is True
    assert f('ab', '.*') is True
    assert f('aab', 'c*a*b') is True
    assert f('mississippi', 'mis*is*p*.') is False
